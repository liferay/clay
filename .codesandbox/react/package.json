{
	"version": "1.0.0",
	"name": "clay-codesandbox",
	"private": true,
	"dependencies": {
		"@clayui/alert": "latest",
		"@clayui/autocomplete": "latest",
		"@clayui/badge": "latest",
		"@clayui/breadcrumb": "latest",
		"@clayui/button": "latest",
		"@clayui/card": "latest",
		"@clayui/charts": "latest",
		"@clayui/color-picker": "latest",
		"@clayui/css": "latest",
		"@clayui/data-provider": "latest",
		"@clayui/date-picker": "latest",
		"@clayui/drop-down": "latest",
		"@clayui/form": "latest",
		"@clayui/icon": "latest",
		"@clayui/label": "latest",
		"@clayui/link": "latest",
		"@clayui/list": "latest",
		"@clayui/loading-indicator": "latest",
		"@clayui/management-toolbar": "latest",
		"@clayui/modal": "latest",
		"@clayui/multi-select": "latest",
		"@clayui/multi-step-nav": "latest",
		"@clayui/nav": "latest",
		"@clayui/navigation-bar": "latest",
		"@clayui/pagination-bar": "latest",
		"@clayui/pagination": "latest",
		"@clayui/panel": "latest",
		"@clayui/popover": "latest",
		"@clayui/progress-bar": "latest",
		"@clayui/shared": "latest",
		"@clayui/slider": "latest",
		"@clayui/sticker": "latest",
		"@clayui/table": "latest",
		"@clayui/tabs": "latest",
		"@clayui/time-picker": "latest",
		"@clayui/tooltip": "latest",
		"react": "16.8.4",
		"react-dom": "16.8.4",
		"react-scripts": "2.1.3"
	},
	"devDependencies": {
		"typescript": "3.3.3"
	},
	"scripts": {
		"start": "react-scripts start"
	}
}
