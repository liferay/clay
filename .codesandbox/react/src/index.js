import React from 'react';
import ReactDOM from 'react-dom';
import '@clayui/css/lib/css/atlas.css';

function App() {
	return (
		<div className="App">
			<h1>Hello CodeSandbox</h1>
			<h2>Add some clay components to see the magic happen!</h2>
		</div>
	);
}

const rootElement = document.getElementById('root');
ReactDOM.render(<App />, rootElement);
