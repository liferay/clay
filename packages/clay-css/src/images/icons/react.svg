<!--
* SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
* SPDX-FileCopyrightText: © 2020 Contributors to the project Clay <https://github.com/liferay/clay/graphs/contributors>
*
* SPDX-License-Identifier: BSD-3-Clause
-->
<svg viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg"><path d="M16 8.006c0-1.041-1.327-2.028-3.362-2.64.47-2.038.26-3.66-.66-4.179a1.438 1.438 0 0 0-.726-.18v-.003c-.897 0-2.071.628-3.259 1.718-1.187-1.083-2.36-1.705-3.258-1.705v.003c-.274 0-.521.058-.737.18-.916.519-1.122 2.137-.649 4.169C1.321 5.984 0 6.968 0 8.006c0 1.041 1.327 2.028 3.362 2.64-.47 2.038-.26 3.66.66 4.179.211.122.459.18.733.18.897 0 2.07-.629 3.258-1.718 1.187 1.083 2.361 1.705 3.258 1.705.274 0 .522-.058.737-.18.917-.519 1.122-2.137.65-4.169C14.678 10.031 16 9.044 16 8.006Zm-4.752-6.287c.15.003.271.032.372.086.444.25.636 1.202.486 2.426a9.73 9.73 0 0 1-.166.942 16.066 16.066 0 0 0-2.071-.349 15.698 15.698 0 0 0-1.357-1.602c1.06-.971 2.058-1.5 2.736-1.503ZM5.11 9.647c.154.256.31.506.47.75a14.93 14.93 0 0 1-1.337-.254c.12-.413.27-.84.44-1.265.134.256.274.512.427.769Zm-.877-3.778c.427-.1.877-.186 1.344-.253a17.649 17.649 0 0 0-.894 1.528 13.57 13.57 0 0 1-.45-1.275Zm.832 2.147a17.028 17.028 0 0 1 1.461-2.503 17.267 17.267 0 0 1 2.942-.003 16.59 16.59 0 0 1 1.468 2.493 17.032 17.032 0 0 1-1.461 2.503 17.251 17.251 0 0 1-2.943.003 16.59 16.59 0 0 1-1.468-2.493Zm5.824 1.621c.15-.256.29-.516.425-.772.176.43.326.859.45 1.275-.428.103-.878.19-1.344.256.16-.246.32-.5.47-.759Zm-.003-3.272a17.37 17.37 0 0 0-.47-.75c.464.068.91.151 1.338.254-.12.413-.27.84-.44 1.265a15.136 15.136 0 0 0-.428-.769Zm-1.99-1.611A20.543 20.543 0 0 0 8 4.73c-.307 0-.61.007-.907.023.294-.375.597-.718.897-1.026.304.308.607.65.907 1.026ZM4.36 1.818a.736.736 0 0 1 .375-.083v-.003c.676 0 1.677.529 2.74 1.493a15.197 15.197 0 0 0-1.347 1.599c-.737.077-1.435.196-2.074.353a9.516 9.516 0 0 1-.17-.93c-.153-1.224.036-2.176.476-2.429Zm-.82 8.136c-.323-.1-.63-.208-.914-.327C1.471 9.144.724 8.51.724 8.006c0-.503.747-1.14 1.901-1.621.28-.119.587-.225.904-.324.186.628.43 1.282.734 1.951-.3.667-.542 1.317-.724 1.942Zm1.21 4.339a.79.79 0 0 1-.37-.086c-.443-.25-.635-1.202-.485-2.426.036-.301.095-.619.166-.942.64.153 1.337.272 2.071.349.44.593.897 1.131 1.357 1.602-1.063.971-2.061 1.503-2.74 1.503Zm3.258-2.01A13.3 13.3 0 0 1 7.1 11.26c.293.012.593.022.897.022.306 0 .61-.007.906-.023-.293.375-.596.718-.896 1.026Zm3.633 1.91a.736.736 0 0 1-.375.084c-.676 0-1.677-.529-2.74-1.493.457-.471.913-1.006 1.347-1.6a15.478 15.478 0 0 0 2.074-.352c.075.324.134.635.17.933.153 1.224-.036 2.175-.476 2.429Zm1.731-4.566c-.28.119-.587.225-.903.324A15.22 15.22 0 0 0 11.734 8c.3-.667.542-1.317.724-1.942.323.1.63.208.917.327 1.154.483 1.901 1.118 1.901 1.621-.003.503-.75 1.14-1.905 1.621Z"/><path d="M7.997 9.47c.823 0 1.49-.655 1.49-1.464 0-.809-.667-1.464-1.49-1.464s-1.49.655-1.49 1.464c0 .809.667 1.464 1.49 1.464Z"/></svg>
