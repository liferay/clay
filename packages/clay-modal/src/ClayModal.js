import 'clay-button';
import 'clay-icon';
import {Config} from 'metal-state';
import {EventHandler} from 'metal-events';
import Component from 'metal-component';
import defineWebComponent from 'metal-web-component';
import dom from 'metal-dom';
import Soy from 'metal-soy';

import templates from './ClayModal.soy.js';

const KEY_CODE_ESC = 27;

/**
 * Metal ClayModal component.
 */
class ClayModal extends Component {
	/**
	 * @inheritDoc
	 */
	created() {
		this._overlayElement = this._valueOverlayElementFn();
		this._eventHandler = new EventHandler();
	}

	/**
	 * @inheritDoc
	 */
	attached() {
		this.addListener('click', this._handleDocumentClick.bind(this), true);
		this.addListener('hide', this._defaultHideModal, true);
		this.addListener('show', this._defaultShowModal, true);
		this.addListener(
			'touchend',
			this._handleDocumentClick.bind(this),
			true
		);
		this.addListener('transitionend', this._handleTransitionEnd, true);
	}

	/**
	 * @inheritDoc
	 */
	detached() {
		super.detached();
		this._eventHandler.removeAllListeners();
	}

	/**
	 * @inheritDoc
	 */
	// eslint-disable-next-line
	sync_isTransitioning() {
		if (this._isTransitioning && !this.visible) {
			this._isTransitioning = false;
			this.visible = true;
		} else if (this._isTransitioning && this.visible) {
			this.visible = false;
		}
	}

	/**
	 * @inheritDoc
	 */
	syncVisible() {
		if (this.visible) {
			dom.enterDocument(this._overlayElement);
			this._overlayElement.offsetHeight;
			dom.addClasses(this._overlayElement, 'show');
		} else {
			dom.exitDocument(this._overlayElement);
		}
	}

	/**
	 * Close modal and remove transitions.
	 * @private
	 */
	_defaultHideModal() {
		dom.removeClasses(this._overlayElement, 'show');
		dom.removeClasses(document.body, 'modal-open');

		this._isTransitioning = true;

		this._eventHandler.removeAllListeners();
	}

	/**
	 * Open modal and add transition.
	 * @private
	 */
	_defaultShowModal() {
		dom.addClasses(document.body, 'modal-open');

		this._isTransitioning = true;

		this._eventHandler.add(
			dom.on(document, 'keyup', this._handleKeyup.bind(this))
		);
	}

	/**
	 * Handle the click buttons and emit event.
	 * @param {!Event} event
	 * @private
	 */
	_handleClickFooterButton(event) {
		this.emit('clickButton', event);
	}

	/**
	 * Handle the footer close button click and emits events.
	 * @param {!Event} event
	 * @private
	 */
	_handleClickCloseButtonFooter(event) {
		this.emit('clickButton', event);
		this.emit('hide');
	}

	/**
	 * Handle the click and emit close modal.
	 * @private
	 */
	_handleCloseModal() {
		this.emit('hide');
	}

	/**
	 * Check if contain click target event in element and emit close modal.
	 * @param {!Event} event
	 * @private
	 */
	_handleDocumentClick(event) {
		if (dom.contains(event.target, this.element)) {
			this.emit('hide');
		}
	}

	/**
	 * Handle click key code esc and emit close modal.
	 * @param {!Event} event
	 * @private
	 */
	_handleKeyup(event) {
		if (event.keyCode === KEY_CODE_ESC) {
			this.emit('hide');
		}
	}

	/**
	 * Handle css transition ends.
	 * @param {!Event} event
	 * @private
	 */
	_handleTransitionEnd(event) {
		if (
			event.target === this.element &&
			this._isTransitioning &&
			!this.visible
		) {
			this._isTransitioning = false;
		}
	}

	/**
	 * Create fragment of the overlay modal.
	 * @return {Element} The resulting document fragment.
	 * @private
	 */
	_valueOverlayElementFn() {
		return dom.buildFragment('<div class="modal-backdrop fade"></div>')
			.firstChild;
	}

	/**
	 * Emit open modal.
	 * @public
	 */
	show() {
		this.emit('show');
	}
}

/**
 * State definition.
 * @static
 * @type {!Object}
 */
ClayModal.STATE = {
	/**
	 * Body of the modal.
	 * @instance
	 * @memberof ClayModal
	 * @type {?string|html|undefined}
	 * @default undefined
	 */
	body: Config.any(),

	/**
	 * CSS classes to be applied to the element.
	 * @instance
	 * @memberof ClayModal
	 * @type {?string|undefined}
	 * @default undefined
	 */
	elementClasses: Config.string(),

	/**
	 * Buttons of the footer.
	 * @instance
	 * @memberof ClayModal
	 * @type {?array|undefined}
	 * @default undefined
	 */
	footerButtons: Config.arrayOf(
		Config.shapeOf({
			alignment: Config.oneOf(['left', 'right']).value('right'),
			label: Config.string().required(),
			style: Config.oneOf(['link', 'primary', 'secondary', 'unstyled']),
			type: Config.oneOf(['button', 'close', 'reset', 'submit']).value(
				'button'
			),
		})
	),

	/**
	 * Id to be applied to the element.
	 * @instance
	 * @memberof ClayModal
	 * @type {?string|undefined}
	 * @default undefined
	 */
	id: Config.string(),

	/**
	 * Adds transitions when show is called.
	 * @instance
	 * @memberof ClayModal
	 * @type {?bool}
	 * @default false
	 * @private
	 */
	_isTransitioning: Config.bool()
		.value(false)
		.internal(),

	/**
	 * The size of element modal.
	 * @instance
	 * @memberof ClayModal
	 * @type {?string|undefined}
	 * @default undefined
	 */
	size: Config.oneOf(['full-screen', 'lg', 'sm']),

	/**
	 * The path to the SVG spritemap file containing the icons.
	 * @instance
	 * @memberof ClayModal
	 * @type {?string|undefined}
	 * @default undefined
	 */
	spritemap: Config.string(),

	/**
	 * Status messages.
	 * @instance
	 * @memberof ClayModal
	 * @type {?string|undefined}
	 * @default undefined
	 */
	status: Config.oneOf(['danger', 'info', 'success', 'warning']),

	/**
	 * Title of the modal.
	 * @instance
	 * @memberof ClayModal
	 * @type {?string|undefined}
	 * @default undefined
	 */
	title: Config.string(),

	/**
	 * Url to place an iframe in the body of the modal.
	 * @instance
	 * @memberof ClayModal
	 * @type {?string|undefined}
	 * @default undefined
	 */
	url: Config.string(),

	/**
	 * Modal visible when show is called.
	 * @instance
	 * @memberof ClayModal
	 * @type {?bool}
	 * @default false
	 * @private
	 */
	visible: Config.bool().value(false),
};

defineWebComponent('clay-modal', ClayModal);

Soy.register(ClayModal, templates);

export {ClayModal};
export default ClayModal;
