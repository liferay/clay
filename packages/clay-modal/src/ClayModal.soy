{namespace ClayModal}

/**
 * This renders the component's whole content.
 */
{template .render}
	{@param? _handleClickCloseButtonFooter: any}
	{@param? _handleClickFooterButton: any}
	{@param? _handleCloseModal: any}
	{@param? _isTransitioning: bool}
	{@param? body: string|html}
	{@param? elementClasses: string}
	{@param? footerButtons: list<?>}
	{@param? id: string}
	{@param? size: string}
	{@param? spritemap: string}
	{@param? status: string}
	{@param? title: string}
	{@param? url: string}
	{@param? visible: bool}

	{let $attributes kind="attributes"}
		class="fade modal
			{if $_isTransitioning or $visible}
				{sp}d-block
			{/if}

			{if $visible}
				{sp}show
			{/if}

			{if $elementClasses}
				{sp}{$elementClasses}
			{/if}
		"

		{if $id}
			id="{$id}"
		{/if}

		role="dialog"
		tabindex="-1"
	{/let}

	<div {$attributes}>
		{let $classesSize kind="text"}
			modal-dialog
			{if $size}
				{sp}modal-{$size}
			{/if}

			{if $status}
				{sp}modal-{$status}
			{/if}
		{/let}

		<div class="{$classesSize}">
			<div class="modal-content">
				{call .header}
					{param _handleCloseModal: $_handleCloseModal /}
					{param spritemap: $spritemap /}
					{param status: $status /}
					{param title: $title /}
				{/call}

				{call .body}
						{param content: $body /}
						{param url: $url /}
				{/call}

				{call .footer}
					{param _handleClickCloseButtonFooter: $_handleClickCloseButtonFooter /}
					{param _handleClickFooterButton: $_handleClickFooterButton /}
					{param buttons: $footerButtons /}
				{/call}
			</div>
		</div>
	</div>
{/template}

/**
 * Render a header of the modal.
 */
{template .header}
	{@param? _handleCloseModal: any}
	{@param? spritemap: string}
	{@param? status: string}
	{@param? title: string}

	{if $title}
		<div class="modal-header">
			{if $spritemap and $status}
				{let $icon kind="text"}
					{switch $status}
						{case 'danger'}
							exclamation-full
						{case 'success'}
							check-circle
						{case 'warning'}
							question-circle-full
						{default}
							info-circle
					{/switch}
				{/let}

				<span class="modal-title-indicator">
					{call ClayIcon.render}
						{param spritemap: $spritemap /}
						{param symbol: $icon /}
					{/call}
				</span>
			{/if}

			<div class="modal-title">{$title}</div>

			{if $spritemap}
				{call ClayButton.render}
					{param elementClasses: 'close' /}
					{param events: ['click': $_handleCloseModal] /}
					{param icon: 'times' /}
					{param spritemap: $spritemap /}
					{param style: 'unstyled' /}
				{/call}
			{/if}
		</div>
	{/if}
{/template}

/**
 * Render a body of the modal.
 */
{template .body}
	{@param? content: string|html}
	{@param? url: string}

	{if $url}
		<div class="modal-body modal-body-iframe">
			{call .iframe}
				{param url: $url /}
			{/call}
		</div>
	{elseif $content}
		<div class="modal-body">
			{$content}
		</div>
	{/if}
{/template}

/**
 * Render a footer of the modal.
 */
{template .footer}
	{@param? _handleClickCloseButtonFooter: any}
	{@param? _handleClickFooterButton: any}
	{@param? buttons: list<?>}

	{if $buttons}
		<div class="modal-footer">
			<div class="modal-item-first">
				{foreach $button in $buttons}
					{if $button.alignment == 'left'}
						{call ClayButton.render}
							{param events: ['click': $_handleClickFooterButton] /}
							{param label: $button.label /}
							{param style: $button.style ?: 'secondary' /}
							{param type: not $button.type or $button.type == 'close' ? 'button' : $button.type /}
						{/call}
						{sp}
					{/if}
				{/foreach}
			</div>
			<div class="modal-item-last">
				{foreach $button in $buttons}
					{if not $button.alignment or $button.alignment == 'right'}
						{if $button.type == 'close'}
							{call ClayButton.render}
								{param elementClasses: 'close-modal' /}
								{param events: ['click': $_handleClickCloseButtonFooter] /}
								{param label: $button.label /}
								{param style: 'secondary' /}
								{param type: 'button' /}
							{/call}
						{else}
							{call ClayButton.render}
								{param events: ['click': $_handleClickFooterButton] /}
								{param label: $button.label /}
								{param style: $button.style ?: 'primary' /}
								{param type: $button.type /}
							{/call}
						{/if}
						{sp}
					{/if}
				{/foreach}
			</div>
		</div>
	{/if}
{/template}

/**
 * Renders an iframe.
 */
{template .iframe}
	{@param url: string}

	<iframe src="{$url}" title="{$url}"></iframe>
{/template}
