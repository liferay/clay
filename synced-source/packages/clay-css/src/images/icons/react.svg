<!--
* SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
* SPDX-FileCopyrightText: © 2020 Contributors to the project Clay <https://github.com/liferay/clay/graphs/contributors>
*
* SPDX-License-Identifier: BSD-3-Clause
-->
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
	<path class="lexicon-icon-outline" d="M512,259.8c0-33.9-42.5-66-107.6-85.9c15-66.3,8.3-119.1-21.1-136c-6.8-4-14.6-5.8-23.3-5.8V32c-28.7,0-66.3,20.4-104.3,55.9c-38-35.2-75.6-55.5-104.3-55.5v0.1c-8.8,0-16.7,1.9-23.6,5.8C98.6,55.3,92,107.9,107.2,174C42.3,194,0,226.1,0,259.8c0,33.9,42.5,66,107.6,85.9c-15,66.3-8.3,119.1,21.1,136c6.8,4,14.7,5.8,23.5,5.8c28.7,0,66.3-20.4,104.3-55.9c38,35.2,75.6,55.5,104.3,55.5c8.8,0,16.7-1.9,23.6-5.8c29.3-16.9,35.9-69.6,20.8-135.7C469.7,325.7,512,293.6,512,259.8z M359.9,55.3c4.8,0.1,8.7,1,11.9,2.8c14.2,8.1,20.3,39.1,15.5,78.9c-1.1,9.8-3,20.1-5.3,30.7c-20.5-5-42.8-8.9-66.3-11.4c-14.1-19.3-28.7-36.8-43.4-52.1C306.3,72.6,338.2,55.4,359.9,55.3z M163.5,313.2c4.9,8.3,9.9,16.5,15,24.4c-14.8-2.2-29.1-4.9-42.8-8.2c3.9-13.5,8.7-27.3,14.1-41.2C154.1,296.5,158.6,304.9,163.5,313.2z M135.5,190.3c13.7-3.2,28.1-6,43-8.2c-5.1,8-10.2,16.3-15,24.7c-4.8,8.3-9.3,16.7-13.6,25C144.2,217.8,139.4,203.8,135.5,190.3z M162.1,260.2c6.5-14,13.8-27.9,21.6-41.6c8.1-14.1,16.5-27.4,25.2-39.8c15.6-1.4,31.3-2.1,47.2-2.1c15.8,0,31.5,0.7,47,2c8.7,12.4,17.1,25.7,25.3,39.6c7.9,13.7,15.1,27.5,21.7,41.5c-6.6,14-13.8,27.9-21.6,41.6c-8.1,14.1-16.5,27.4-25.2,39.8c-15.5,1.4-31.3,2.1-47.2,2.1c-15.8,0-31.5-0.7-47-2c-8.7-12.4-17.1-25.7-25.3-39.6C175.9,288,168.6,274.1,162.1,260.2z M348.5,312.9c4.8-8.3,9.3-16.8,13.6-25.1c5.6,14,10.4,27.9,14.4,41.5c-13.7,3.3-28.1,6.2-43,8.3C338.5,329.6,343.7,321.4,348.5,312.9z M348.4,206.5c-4.8-8.3-9.9-16.5-15-24.4c14.8,2.2,29.1,4.9,42.8,8.2c-3.9,13.5-8.7,27.3-14.1,41.2C357.8,223.1,353.3,214.8,348.4,206.5z M284.7,154c-9.4-0.4-19-0.7-28.7-0.7c-9.8,0-19.5,0.2-29,0.7c9.4-12.2,19.1-23.4,28.7-33.4C265.4,130.6,275.1,141.8,284.7,154z M139.5,58.5c3.1-1.9,7.2-2.7,12-2.7v-0.1c21.6,0,53.6,17.2,87.7,48.6c-14.6,15.3-29.2,32.7-43.1,52c-23.6,2.5-45.9,6.4-66.4,11.5c-2.4-10.4-4.2-20.5-5.4-30.2C119.4,97.7,125.4,66.7,139.5,58.5z M113.2,323.2c-10.3-3.2-20.1-6.8-29.2-10.6c-36.9-15.7-60.8-36.4-60.8-52.8s23.9-37.1,60.8-52.8c9-3.9,18.8-7.3,28.9-10.5c5.9,20.4,13.8,41.7,23.5,63.5C126.8,281.7,119.1,302.9,113.2,323.2z M152,464.4c-4.7-0.1-8.7-1-11.8-2.8c-14.2-8.1-20.4-39.1-15.6-78.9c1.1-9.8,3-20.1,5.3-30.7c20.5,5,42.8,8.9,66.3,11.4c14.1,19.3,28.7,36.8,43.4,52.1C205.6,447.1,173.7,464.4,152,464.4z M256.2,399c-9.7-10-19.4-21.2-29-33.4c9.4,0.4,19,0.7,28.7,0.7c9.8,0,19.5-0.2,29-0.7C275.5,377.9,265.8,389,256.2,399z M372.5,461.2c-3.1,1.9-7.2,2.7-12,2.7c-21.6,0-53.6-17.2-87.7-48.6c14.6-15.3,29.2-32.7,43.1-52c23.6-2.5,45.9-6.4,66.4-11.5c2.4,10.5,4.3,20.6,5.4,30.3C392.6,422,386.6,453,372.5,461.2z M427.9,312.6c-9,3.9-18.8,7.3-28.9,10.5c-5.9-20.4-13.8-41.7-23.5-63.5c9.6-21.7,17.3-42.9,23.2-63.2c10.3,3.2,20.1,6.8,29.3,10.6c36.9,15.7,60.8,36.4,60.8,52.8C488.7,276.2,464.8,297,427.9,312.6z"/>
	<path class="lexicon-icon-outline" d="M255.9,307.5c26.3,0,47.7-21.3,47.7-47.7s-21.4-47.7-47.7-47.7c-26.3,0-47.7,21.3-47.7,47.7S229.6,307.5,255.9,307.5z"/>
</svg>