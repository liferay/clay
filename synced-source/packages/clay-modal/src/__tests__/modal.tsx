/**
 * SPDX-FileCopyrightText: Â© 2019 Liferay, Inc. <https://liferay.com>
 * SPDX-License-Identifier: BSD-3-Clause
 */

import ClayModal, {useModal} from '..';
import {cleanup, fireEvent, render} from '@testing-library/react';
import React from 'react';

const spritemap = 'icons.svg';

describe('ClayModal custom opener', () => {
	afterEach(() => {
		cleanup();
	});

	it('renders inside a container element', () => {
		const ModalWithState = () => {
			const {observer} = useModal({onClose: () => {}});
			const containerRef = React.useRef<HTMLDivElement | null>(null);

			return (
				<>
					<div id="container" ref={containerRef}>
						container
					</div>

					<ClayModal
						containerElementRef={containerRef}
						observer={observer}
						spritemap={spritemap}
					/>
				</>
			);
		};

		render(<ModalWithState />);

		expect(document.body).toMatchSnapshot();
	});

	it('does not close automatically with disableAutoClose', () => {
		const handleOnClose = jest.fn();

		const ModalWithState = () => {
			const [visibleModal] = React.useState(true);

			const {observer} = useModal({onClose: handleOnClose});

			return (
				<>
					{visibleModal && (
						<ClayModal
							disableAutoClose
							observer={observer}
							spritemap={spritemap}
						/>
					)}
				</>
			);
		};

		const handleKeyDown = jest.fn();

		document.body.addEventListener('keydown', handleKeyDown);

		render(<ModalWithState />);

		fireEvent.keyDown(document.body, {
			charCode: 0,
			code: 'Escape',
			key: 'Escape',
		});

		expect(document.body).toMatchSnapshot();

		expect(handleKeyDown).toHaveBeenCalled();

		expect(handleOnClose).not.toHaveBeenCalled();

		document.body.removeEventListener('keydown', handleKeyDown);
	});
});
