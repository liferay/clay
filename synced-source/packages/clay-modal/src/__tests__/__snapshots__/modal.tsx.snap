// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayModal custom opener does not close automatically with disableAutoClose 1`] = `
<body
  class="modal-open"
>
  <div />
  <div>
    <div
      aria-hidden="true"
      class="modal-backdrop fade"
    />
    <div
      class="fade modal d-block"
    >
      <div
        class="modal-dialog"
      >
        <div
          aria-labelledby="clay-modal-label-2"
          aria-modal="true"
          class="modal-content"
          role="dialog"
          tabindex="-1"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`ClayModal custom opener renders inside a container element 1`] = `
<body
  class="modal-open"
>
  <div>
    <div
      id="container"
    >
      container
      <div>
        <div
          aria-hidden="true"
          class="modal-backdrop fade"
        />
        <div
          class="fade modal d-block"
        >
          <div
            class="modal-dialog"
          >
            <div
              aria-labelledby="clay-modal-label-1"
              aria-modal="true"
              class="modal-content"
              role="dialog"
              tabindex="-1"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
